File: Ruby/challenge1.rb


def beginner a
	return "Has #{a.length} and a sum of #{a.sum}"
end

def advanced a, b
	return (a**b).to_s.split('').inject (0){|sum, n| sum + n.to_i}
end

puts beginner [1, 2, 3, 4, 5, 6]
puts advanced 2, 5
puts advanced 3, 10

File: Python/challenge1.py
import functools as ft

def beginner(a):
	return f"Has length: {len(a)} and sum {sum(a)}"

def advanced(a, b):
	return ft.reduce(sum, list(str(a**b)))

def sum(lis):
	return int(aSum) + int(n)

print(advanced(2, 5))
print(advanced(3, 10))
